File: app/src/main/java/com/example/zs_wan_android/main/home/HomeContract.java
package com.example.zs_wan_android.main.home;

import com.example.zs_wan_android.base.IBasePresenter;
import com.example.zs_wan_android.base.IBaseView;

public interface HomeContract {
    interface View extends IBaseView {
        void showData();
    }

    interface Presenter<T> extends IBasePresenter<HomeContract.View>{
        void loadData();
    }
}

File: app/src/main/java/com/example/zs_wan_android/main/home/HomeContract.kt
package com.example.zs_wan_android.main.home

import com.example.zs_wan_android.base.IBasePresenter
import com.example.zs_wan_android.base.IBaseView
import com.example.zs_wan_android.entity.BannerEntity
import com.example.zs_wan_android.entity.HomeEntity

interface HomeContract {
    interface View : IBaseView {
        fun showList(list: MutableList<HomeEntity.DatasBean>)
        fun showBanner(bannerList:MutableList<BannerEntity>)
    }

    interface Presenter<T> : IBasePresenter<View> {
        fun loadData(pageNum:Int)
        fun loadBanner()
    }
}